test/MOI/MOI_wrapper.jl

```diff
@@ -1177,6 +1173,58 @@ function test_indicator_nonconstant()
     @test_throws ErrorException MOI.add_constraint(model, f, s)
 end
 
+function test_farkas_dual_min()
+    MOI.empty!(OPTIMIZER)
+    x = MOI.add_variables(model, 2)
+    MOI.set(model, MOI.ObjectiveSense(), MOI.MIN_SENSE)
+    MOI.set(
+        model,
+        MOI.ObjectiveFunction{MOI.SingleVariable}(),
+        MOI.SingleVariable(x[1]),
+    )
+    clb = MOI.add_constraint.(
+        model, MOI.SingleVariable.(x), MOI.GreaterThan(0.0)
+    )
+    c = MOI.add_constraint(
+        model,
+        MOI.ScalarAffineFunction(MOI.ScalarAffineTerm.([2.0, 1.0], x), 0.0),
+        MOI.LessThan(-1.0),
+    )
+    MOI.optimize!(model)
+    @test MOI.get(model, MOI.TerminationStatus()) == MOI.INFEASIBLE
+    @test MOI.get(model, MOI.DualStatus()) == MOI.INFEASIBILITY_CERTIFICATE
+    clb_dual = MOI.get.(model, MOI.ConstraintDual(), clb)
+    c_dual = MOI.get(model, MOI.ConstraintDual(), c)
+    @test clb_dual[1] ≈ -2 * c_dual atol = 1e-6
+    @test clb_dual[2] ≈ -c_dual atol = 1e-6
+end
+
+function test_farkas_dual_max()
+    MOI.empty!(OPTIMIZER)
+    x = MOI.add_variables(model, 2)
+    MOI.set(model, MOI.ObjectiveSense(), MOI.MAX_SENSE)
+    MOI.set(
+        model,
+        MOI.ObjectiveFunction{MOI.SingleVariable}(),
+        MOI.SingleVariable(x[1]),
+    )
+    clb = MOI.add_constraint.(
+        model, MOI.SingleVariable.(x), MOI.GreaterThan(0.0)
+    )
+    c = MOI.add_constraint(
+        model,
+        MOI.ScalarAffineFunction(MOI.ScalarAffineTerm.([2.0, 1.0], x), 0.0),
+        MOI.LessThan(-1.0),
+    )
+    MOI.optimize!(model)
+    @test MOI.get(model, MOI.TerminationStatus()) == MOI.INFEASIBLE
+    @test MOI.get(model, MOI.DualStatus()) == MOI.INFEASIBILITY_CERTIFICATE
+    clb_dual = MOI.get.(model, MOI.ConstraintDual(), clb)
+    c_dual = MOI.get(model, MOI.ConstraintDual(), c)
+    @test clb_dual[1] ≈ -2 * c_dual atol = 1e-6
+    @test clb_dual[2] ≈ -c_dual atol = 1e-6
```

Comment:
@blegat, it's not obvious to me what our sign conventions are.

Currently for `MIN_SENSE`, Gurobi returns
```Julia
(clb_dual, c_dual) = ([2.0, 1.0], -1.0)
```
and for `MAX_SENSE`, it returns
```julia
(clb_dual, c_dual) = ([-1.0, -0.5], 0.5)
```
Should the `MAX_SENSE` values be negated?

Reply:
I've negated the `MAX_SENSE` case. The latest commit returns `(clb_dual, c_dual) = ([1.0, 0.5], -0.5)`. I think this makes sense from our definition of duality, but I'm not sure it is what users will expect.

Reply:
Hmm. This might be a bug in Gurobi.
```Julia
julia> function test_farkas(c1)
           env = Ref{Ptr{Cvoid}}()
           ret = GRBloadenv(env, C_NULL)
           GRBsetintparam(env[], "InfUnbdInfo", 1)
           GRBsetintparam(env[], "OutputFlag", 0)
           model = Ref{Ptr{Cvoid}}()
           GRBloadmodel(
               env[],  # *env
               model,  # **modelp
               C_NULL,  # *Pname
               Cint(2),  # numvars
               Cint(1),  # numconstrs
               Cint(-1),  # objsense (maximization)
               0.0, # objcon
               Cdouble[c1, 0.0],  # *obj
               Cchar[GRB_LESS_EQUAL],  # *sense
               Cdouble[-1.0], # *rhs
               Cint[0, 1], # *vbeg
               Cint[1, 1],  # vlen
               Cint[0, 0],  # *vind
               Cdouble[-2.0, -1.0],  # *vval
               Cdouble[-GRB_INFINITY, -GRB_INFINITY],  # *lb
               Cdouble[0.0, 0.0],  # *ub
               C_NULL,  # *vtype
               C_NULL,  # **varnames
               C_NULL,  # **constrnames
           )
           GRBoptimize(model[])
           dual = Vector{Cdouble}(undef, 1)
           GRBgetdblattrarray(model[], "FarkasDual", 0, 1, dual)
           dual
       end
test_farkas (generic function with 1 method)

julia> test_farkas(1.0)
Academic license - for non-commercial use only
1-element Array{Float64,1}:
 1.0

julia> test_farkas(-1.0)
Academic license - for non-commercial use only
1-element Array{Float64,1}:
 -0.5
```

Would we expect the Farkas multiplier to switch based on the sign of an objective coefficient?

It seems to contradict their documentation:
https://www.gurobi.com/documentation/9.0/refman/farkasdual.html#attr:FarkasDual

<img width="891" alt="image" src="https://user-images.githubusercontent.com/8177701/97521644-e1791e00-1a02-11eb-83b9-e797cc944f55.png">

Reply:
Seems to happen only with upper-bounded variables, so this might be a presolve transform leaking through. I've put in a support request.

Reply:
Isn't it `@test clb_dual[2] ≈ -c_dual - 1` to account for the objective?

Reply:
I guess it depends if you do `A' y >= 0`, or `A' y >= b` (i.e., Farkas, or a unbounded ray). 

You previously suggested it was just `club_dual[2] == -c_dual`:
https://github.com/jump-dev/MathOptInterface.jl/pull/127#issuecomment-329997053

The documentation for `INFEASIBILITY_CERTIFICATE` just says that it is a proof of infeasibility, but doesn't specify more:
https://github.com/jump-dev/MathOptInterface.jl/blob/7543afe4b5151cf36bbd18181c1bb5c83266ae2f/src/attributes.jl#L1339-L1340

Reply:
Indeed, I missed that it was infeasible